solutions/usaco-964.mdx

```diff
@@ -2,16 +2,52 @@
 id: usaco-964
 source: USACO Bronze 2019 December
 title: Where Am I?
-author: Ananth Kashyap
+author: Ananth Kashyap, Brad Ma
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_whereami_bronze_dec19.html)
+[Official Analysis(C++)](http://www.usaco.org/current/data/sol_whereami_bronze_dec19.html)
 
 ## Implementation
 
 **Time Complexity:** $\mathcal{O}(N^3)$
 
 <LanguageSection>
+	
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class WhereAmI {
+	public static void main (String[] args) throws IOException {
+		Kattio io = new Kattio("whereami");
+		int n = io.nextInt();
+		String s = io.next();
+
+		for (int guess = 1; guess <= n; guess++) { // try each length
+			boolean good = true;
+			// try all substring combinations of that length
+			for (int i = 0; i + guess <= n; i++) {
+				for (int j = 0; j < i; j++) {
+					if (s.substring(i, i + guess).equals(s.substring(j, j + guess))) {
```

Comment:
how many chars is this, it seems a bit long

Reply:
^ it's 87 characters

Reply:
does the 87 include the indents in the front?

Reply:
Yeah

Reply:
I would recommend storing the two substrings in variables